mogp_gpu/src/cov_gpu.cu

```diff
@@ -0,0 +1,182 @@
+#include "cov_gpu.hpp"
+#include <stdio.h>
+
+// Covariance device function
+__device__ REAL cov_val_d(int Ninput, REAL *x_d, REAL *y_d, REAL *theta_d)
+{
+    REAL s = 0.0;
+    for (unsigned int i=0; i < Ninput; i++)
+    {
+        REAL d_i = x_d[i] - y_d[i];
+        s += d_i * d_i * exp(theta_d[i]);
+    }
+    return exp(-0.5 * s + theta_d[Ninput]);
```

Comment:
Is the nugget added here for fitted nugget values? I didn't see it added in the main class, but I could be missing something.

Reply:
This was indeed missing - now implemented (see above comment for details).